core/indexer/processTransactions.go

```diff
@@ -224,20 +226,23 @@ func (tdp *txDatabaseProcessor) groupNormalTxsAndRewards(
 			txs := getTransactions(txPool, mb.TxHashes)
 			for hash, tx := range txs {
 				dbTx := tdp.commonProcessor.buildTransaction(tx, []byte(hash), mbHash, mb, header, mbTxStatus)
+				addToAlteredAddresses(dbTx, alteredAddresses, mb, selfShardID)
 				transactions[hash] = dbTx
 				delete(txPool, hash)
 			}
 		case block.InvalidBlock:
 			txs := getTransactions(txPool, mb.TxHashes)
 			for hash, tx := range txs {
 				dbTx := tdp.commonProcessor.buildTransaction(tx, []byte(hash), mbHash, mb, header, txStatusInvalid)
+				addToAlteredAddresses(dbTx, alteredAddresses, mb, selfShardID)
 				transactions[hash] = dbTx
 				delete(txPool, hash)
 			}
 		case block.RewardsBlock:
 			rTxs := getRewardsTransaction(txPool, mb.TxHashes)
 			for hash, rtx := range rTxs {
 				dbTx := tdp.commonProcessor.buildRewardTransaction(rtx, []byte(hash), mbHash, mb, header, mbTxStatus)
+				alteredAddresses[dbTx.Receiver] = struct{}{}
```

Comment:
might consider calling the same addToAlteredAddresses as to keep the same order of abstraction in the function

Reply:
done